pkg/cmd/get/klusterletinfo/exec.go

```diff
@@ -0,0 +1,245 @@
+	}
+	imageName := "<none>"
+	registrationOperatorExpectedRs := *deploy.Spec.Replicas
+	registrationOperatorAvailableRs := deploy.Status.AvailableReplicas
+	for _, container := range deploy.Spec.Template.Spec.Containers {
+		if container.Name == "registration-operator" {
+			imageName = container.Image
+		}
+	}
+	crdStatus := make(map[string]string)
+	cmgrCrd, err := o.crdClient.ApiextensionsV1().
+		CustomResourceDefinitions().
+		Get(context.TODO(), klusterletCRD, metav1.GetOptions{})
+	if err != nil {
+		if !apierrors.IsNotFound(err) {
+			return err
+		}
+	}
+	if cmgrCrd != nil {
+		crdStatus[klusterletCRD] = "installed"
+	} else {
+		crdStatus[klusterletCRD] = "absent"
+	}
+
+	o.printer.Write(printer.LEVEL_0, "Registration Operator:\n")
+	o.printer.Write(printer.LEVEL_1, "Controller:\t(%d/%d) %s\n", registrationOperatorAvailableRs, registrationOperatorExpectedRs, imageName)
+	o.printer.Write(printer.LEVEL_1, "CustomResourceDefinition:\n")
+	for name, st := range crdStatus {
+		o.printer.Write(printer.LEVEL_2, "(%s) %s\n", st, name)
+	}
+	return nil
+}
+
+func (o *Options) printComponents() error {
+	klet, err := o.operatorClient.OperatorV1().
+		Klusterlets().
+		Get(context.TODO(), klusterletName, metav1.GetOptions{})
+	if err != nil {
+		if !apierrors.IsNotFound(err) {
+			return err
+		}
+		o.printer.Write(printer.LEVEL_0, "Components:\t<uninstalled>\n")
+		return nil
+	}
+
+	o.printer.Write(printer.LEVEL_0, "Components:\n")
+
+	if err := o.printRegistration(klet); err != nil {
+		return err
+	}
+	if err := o.printWork(klet); err != nil {
+		return err
+	}
+	if err := o.printComponentsCRD(klet); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (o *Options) printRegistration(klet *v1.Klusterlet) error {
+	klusterletAgent, err := o.kubeClient.AppsV1().
+		Deployments(componentsNamespace).
+		Get(context.TODO(), componentNameRegistrationAgent, metav1.GetOptions{})
+	if err != nil {
+		if !apierrors.IsNotFound(err) {
+			return err
+		}
+		return nil
+	}
+
+	controllerExpectedRs := int(*klusterletAgent.Spec.Replicas)
+	controllerAvailableRs := klusterletAgent.Status.AvailableReplicas
+
+	o.printer.Write(printer.LEVEL_1, "Registration:\n")
+	o.printer.Write(printer.LEVEL_2, "Agent:\t(%d/%d) %s\n", controllerAvailableRs, controllerExpectedRs, getImageName(klusterletAgent))
+	return nil
+}
+
+func (o *Options) printWork(klet *v1.Klusterlet) error {
+	workAgent, err := o.kubeClient.AppsV1().
+		Deployments(componentsNamespace).
+		Get(context.TODO(), componentNameWorkAgent, metav1.GetOptions{})
+	if err != nil {
+		if !apierrors.IsNotFound(err) {
+			return err
+		}
+		return nil
+	}
+
+	agentExpectedRs := int(*workAgent.Spec.Replicas)
+	agentAvailableRs := workAgent.Status.AvailableReplicas
+	o.printer.Write(printer.LEVEL_1, "Work:\n")
+	o.printer.Write(printer.LEVEL_2, "Webhook:\t(%d/%d) %s\n", agentExpectedRs, agentAvailableRs, getImageName(workAgent))
+	return nil
+}
+
+func (o *Options) printComponentsCRD(klet *v1.Klusterlet) error {
```

Comment:
should this be a common utility for printers that could be shared for hub and spoke printer